src/bin/leftwm-worker.rs

```diff
@@ -91,6 +96,13 @@ async fn event_loop(
                 event_buffer.append(&mut get_events(display_server));
                 continue;
             }
+            //Once in a blue moon we miss the focus event,
+            //This is to double check that we know which window is currently focused
+            _ = timeout(500), if event_buffer.is_empty() => {
```

Comment:
There does appear to be a bit of delay in focusing the new window: (sorry it's blurry, forgot to turn the compositor off!)
![Peek 2021-03-31 20-20](https://user-images.githubusercontent.com/2220164/113230598-f6323a00-925e-11eb-9b16-592e4838b757.gif)

I wonder if 500ms is the right amount of time? I know I probably wouldn't notice, but might miss a keystroke or something.

Reply:
Yeah I feel 500ms is too high for an interactive system, for my testing I have lowered this to 250ms to see if it still works completely and will lower it as far as it can on my setup (ideally 100 or below)

Reply:
Would something like this work to find out the optimum value
```
            _ = timeout(25), if event_buffer.is_empty() => {
                for n in 0..50 {
                    let mut focus_event = display_server.verify_focused_window();
                    if !focus_event.is_empty() {
                        event_buffer.append(&mut focus_event);
                        if n > 0 {
                            log::info!("Leftwm n: {:?}", n);
                        }
                        break;
                    }
                    timeout(5);
                }
                continue;
            }
```

Reply:
@mautamu can you check where n gets to in your use cases, as with my testing (where issues occurred before) seem to be fixed by the other changes outside the delay so n doesn't go above 0 with even 0 delay.